### PR TITLE
Escape room room and chests

### PR DESCRIPTION
**Pull request by Jasperkroes**
Added a Chest, Room and Progress object.
Controller is split into multiple controllers. 

Chest:
img_20180531_111410

Room:
img_20180531_111420

Progress:
img_20180531_111434

Controller:
img_20180531_111446

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -145,10 +145,9 @@
         <!-- Miscellaneous other checks.                   -->
         <!-- See http://checkstyle.sf.net/config_misc.html -->
         <module name="ArrayTypeStyle"/>
-        <module name="FinalParameters"/>
```
Comment (lisv) [inline checkstyle.xml]: no

File: files/standard.json [hunk 2]
```diff
@@ -9,7 +9,21 @@
...
-    "chests": 9
+    "people": 5,
+    "chests": [
+      {
+        "sections": 2,
+        "targetDuration": 120
+      },
+      {
+        "sections": 3,
+        "targetDuration": 220
```
Comment (lisv) [inline files/standard.json]: 220? mins? thats a lot since most escape last an hour

**Comment by lisv** [comment]
What happens when an incorrect configuration file is chosen, or a correct configuration file with missing values? E.g. values the host is not sure about yet

File: files/standard.json (hunk 2 above)
Comment (egedebruin) [reply files/standard.json]: It is seconds, but maybe an explanation of the jsonfile could be useful.

File: checkstyle.xml (hunk 1 above)
Comment (Jasperkroes) [reply checkstyle.xml]: java 8 zet alle variabelen automatisch op final, dus dit is onnodig

File: files/standard.json (hunk 2 above)
Comment (Jasperkroes) [reply files/standard.json]: duration is in seconden

**Comment by egedebruin** [comment]
@lisv Now a NullPointerException will be thrown when an incorrect config file is chosen or when values are missing. I can change that a default value will be used when values are missing.

File: src/main/java/gui/BaseScene.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package gui;
 
+import gui.controllers.Controller;
```
Comment (robinjo78) [inline src/main/java/gui/BaseScene.java]: Why do we need this here, where is it used?

File: src/main/java/gui/controllers/Controller.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        streamStage.close();
+        cameraHandler.addCamera(streamUrl);
+    }
+
+    /**
+     * Method to initialize a connection with our active camera(stream).
+     *
+     * @param streamUrl THE url
+     */
+    public void createTheStream(final String streamUrl) {
```
Comment (robinjo78) [inline src/main/java/gui/controllers/Controller.java]: Still need this method?

File: checkstyle.xml (hunk 1 above)
Comment (egedebruin) [reply checkstyle.xml]: bron?

Comment (lisv) [reply checkstyle.xml]: Ja bron?

File: files/standard.json (hunk 2 above)
Comment (lisv) [reply files/standard.json]: Is minutes not better? Should the host "guess" how much seconds it takes? Seems too specific for this kinda thing

Comment (egedebruin) [reply files/standard.json]: How is it guessing? And I think seconds isn't too specific, for when you want to have the half of a minute.

File: src/main/java/gui/controllers/TimeLogController.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                addInformation(log);
+            }
+        }
+    }
+
+    /**
+     * Turns the button invisible after it is clicked.
+     */
+    public void confirmedChest() {
+        addInformation("Found chest.");
```
Comment (robinjo78) [inline src/main/java/gui/controllers/TimeLogController.java]: Why has the dot reappeared?

File: checkstyle.xml (hunk 1 above)
Comment (Jasperkroes) [reply checkstyle.xml]: https://docs.oracle.com/javase/specs/jls/se8/html/jls-4.html#jls-4.12.4

Comment (Jasperkroes) [reply checkstyle.xml]: https://stackoverflow.com/questions/20938095/difference-between-final-and-effectively-final

**Comment by robinjo78** [comment]
The last used directory should be shown when choosing a json file again.

**Comment by egedebruin** [comment]
@robinjo78 The application now gives a warning at start stream when no cameras are known to be used.

**Comment by lisv** [comment]
Also, I think that maybe the name of Controller should be changed, cause of right now it is not really clear what it does (without looking at all its methods)

**Comment by egedebruin** [comment]
@lisv Config issue is fixed.

**Comment by egedebruin** [comment]
ok everyting fixed

**Pull request merged**